databricks/koalas/frame.py

```diff
@@ -7830,43 +7830,86 @@ def reindex(
 
     def _reindex_index(self, index, fill_value):
         # When axis is index, we can mimic pandas' by a right outer join.
-        assert (
-            len(self._internal.index_spark_column_names) <= 1
-        ), "Index should be single column or not set."
+        nlevels = len(self._internal.index_spark_column_names)
+        assert nlevels <= 1 or (
+            isinstance(index, ks.MultiIndex) and nlevels == index.nlevels
+        ), "MultiIndex DataFrame can only be reindexed with a similar Koalas MultiIndex."
 
-        index_column = self._internal.index_spark_column_names[0]
-
-        kser = ks.Series(list(index))
-        labels = kser._internal.spark_frame.select(kser.spark.column.alias(index_column))
+        index_columns = self._internal.index_spark_column_names
         frame = self._internal.resolved_copy.spark_frame.drop(NATURAL_ORDER_COLUMN_NAME)
 
+        if isinstance(index, ks.Index):
+            if nlevels != index.nlevels:
+                # Use temporary column so we can call reindex in order to fill columns on new index
```

Comment:
nit: do you want to remove or rephrase this comment?

Reply:
Yes I removed it